### ISSUE TITLE
Zigbee lights always being turned off

### ISSUE THREAD

**Issue by scottydj2** [domoticz/domoticz#5709]
Hi,

I'm currently running a docker stack on an RPI 4B containing mosquitto, zigbee2mqtt and domoticz. 
I noticed that when I upgraded from 2022.2 to 2023.1, I was no longer able to properly control my zigbee lights (collection of single and 2 channel dimmers. Tuya_601s, MOES 105B). Further investigation revealed that both on and off commands were being picked up by Z2MQTT as state off. 
Following debugging has been done on the latest beta code running on my windows dev PC (as it's easier to debug), with mosquitto and zigbee2mqtt still running on the RPI.
 
From the domoticz gui, when I click the button to turn on my office light 1 which was off, I get the following in the domoticz log:
 2023-06-04 23:38:39.455 MQTT Gateway: MQTT received: Scott's Office Lights l1 (value: {"brightness":107,"countdown":null,"state":"OFF"})
2023-06-04 23:38:39.563 MQTT Gateway: MQTT received: Scott's Office Lights l2 (value: {"brightness":130,"countdown":null,"state":"ON"})
2023-06-04 23:38:39.137 Status: User: admin (IP: 192.168.11.15) initiated a switch command (5/Scott's Office Lights l1/On) 

and the following in the Z2MQTT log:

info 2023-06-04 23:38:39MQTT publish: topic 'zigbee2mqtt/Scott's Office Lights', payload '{"brightness_l1":107,"brightness_l2":130,"countdown_l1":null,"countdown_l2":null,"linkquality":61,"state_l1":"OFF","state_l2":"ON"}'
info 2023-06-04 23:38:39MQTT publish: topic 'zigbee2mqtt/Scott's Office Lights/l1', payload '{"brightness":107,"countdown":null,"state":"OFF"}'
info 2023-06-04 23:38:39MQTT publish: topic 'zigbee2mqtt/Scott's Office Lights/l2', payload '{"brightness":130,"countdown":null,"state":"ON"}'

Basically, the 'ON' request is somehow being converted to an off request. After much picking through code, I found the following line in MQTTAutoDiscover.cpp which had been commented out as part of #5440 .

	```
	3504	else
	3505	{
	3506			root["brightness"] = slevel;
	3507			//root["state"] = (slevel > 0) ? "ON" : "OFF";
	3508	}


Uncommenting line 3507 fixed the problem for me.

Not sure how it's working for the person in the PR though as he was using the same dimmer.

Does no-one else use zigbee dimmers??

Thanks,

Scott

**Comment by gizmocuz** [comment]
Sure, every zigbee light I own can be dimmed (normal bulbs and RGBW)
No problem with them
So you only have this device that is not working?

**Comment by scottydj2** [comment]
I have 15 modules not working :(
All are identified  as Tuya TS0601_dimmer_1 or Tuya TS0601_dimmer_2.
I have a couple of others Lonsonho and Tuya TS110E which seem to still work OK.

**Comment by gizmocuz** [comment]
And did you update your zigbee2mqtt as well? I also cannot see that setting brightness value needs a on off state value as well
Anyway, you upgraded to the stable version 2023.1 correct? Did you also tried with the latest beta version?
(New stable 2023.2 is released in a few minutes, hope that one is working for you, else we need to make a patch in the upcoming beta version for this. But surely (next to mine and his working), there should be a lot of people having this issue...

**Comment by scottydj2** [comment]
Zigbee2mqtt is latest dev 1.32.1-dev, coordinator 20220219.
I have run with 2023.1 stable and currently run latest beta. Both have the problem. 
The original change (#5440) was made before 2023.1 (that's when it broke for me) and only 1 person seemed to be having a problem then.
Guess no-one cares till something breaks?

**Comment by gizmocuz** [comment]
Happy to merge, but could you also ask on the Forum? Because this user had a problem when we also send the state... 
Both work find for my lights

**Comment by scottydj2** [comment]
I'm happy to ask on the forum, but unless this change is available (in a dev branch?) for them to test, not sure how that will help?

**Comment by gizmocuz** [comment]
Okey, so at the moment we have the stable release without the state option (2023.2) and we have a new beta version, with the state option (your PR)
I am curious if the beta works for everyone or the stable or both.
Maybe we need to conclude that this is specific for these Tuya lights, and maybe this patch should be made in zigbee2mqtt as when you set a brightness you would not expect to be sending also the state object. A brightness other than 0 already means On>

**Comment by gizmocuz** [comment]
In the current beta (15614), the state is not set when setting the brightness

**Comment by scottydj2** [comment]
OK, sorry for not getting back on this. I had the stable 2023.2 build runnning and the dimmers were all working OK so thought the PR had gone into that build (this was build 15459) as discussed here [https://www.domoticz.com/forum/viewtopic.php?p=304840#p304840](url).
I'm generally using the docker package builds on a pi4. Today having seen the last comment, I pulled beta 15614, and my dimmers are broken again. Went back to stable stable 2023.2 which worked before and it is now also broken! Tried beta 15607 and it works again. 
So, my system definitely needs that state line in. 
Guess me and @Elmervc need a debugging session.

**Comment by gizmocuz** [comment]
@scottydj2 , could you post your config payload? Let's see if we might discover why the state is needed.
Maybe it could also be solved on the zigbee2mqtt site in the converter

**Comment by scottydj2** [comment]
Hi, I can if you can explain how/where I find/obtain it.

**Comment by gizmocuz** [comment]
Sure, please download/use MQTT Explorer
Connect to your MQTT broker
You will see the auto discovery topic (like homeassistant)
In here you see a sub tree called 'lights', in here you have to find your light like

The topic could be

homeassistant/light/0x00124b0022d36b6d/light/config

(please post it here)

If you click on the config tree item, on the right you will see it's JSON payload, please post that here as well

**Comment by scottydj2** [comment]
OK, for one of the light modules, I have:
homeassistant/light/0xcc86ecfffefa929a/light_l1/config

{
  "availability": [
    {
      "topic": "zigbee2mqtt/bridge/state"
    }
  ],
  "brightness": true,
  "brightness_scale": 254,
  "command_topic": "zigbee2mqtt/Scott's Office Lights/l1/set",
  "device": {
    "identifiers": [
      "zigbee2mqtt_0xcc86ecfffefa929a"
    ],
    "manufacturer": "TuYa",
    "model": "2 gang smart dimmer (TS0601_dimmer_2)",
    "name": "Scott's Office Lights"
  },
  "json_attributes_topic": "zigbee2mqtt/Scott's Office Lights/l1",
  "name": "L1",
  "object_id": "scott's_office_lights_l1",
  "origin": {
    "name": "Zigbee2MQTT",
    "sw": "1.33.1-dev",
    "url": "https://www.zigbee2mqtt.io"
  },
  "schema": "json",
  "state_topic": "zigbee2mqtt/Scott's Office Lights/l1",
  "unique_id": "0xcc86ecfffefa929a_light_l1_zigbee2mqtt"
}

**Comment by gizmocuz** [comment]
@scottydj2 , thank you for your config. Unfraternally, I don't see a specific field that could indicate that it needs the state value.
I still think it should be solved upstream in zigbee2mqtt.
For now, in beta 15640, I check the device manufacturer and if it's Toya, I add the State field.
So it should work again...

**Comment by scottydj2** [comment]
Thanks. I will check the beta tonight.

What I was seeing when debugging originally was that the state field was always being supplied in the command sent from domoticz to zigbee2mqtt. It was just always defaulting to off in the failing case. Are you saying that (from the config) we should not expect the state field to be supplied?

**Comment by gizmocuz** [comment]
No, I'm not saying that. 
But when setting the brightness value (like 60%), we also have to supply the 'State' value (on/off, on=brightness>0)

**Comment by scottydj2** [comment]
I can confirm beta 15640 is working fine for me. Thanks.
